.github/workflows/clean-aws-resources.yml

```diff
@@ -0,0 +1,46 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: Monthly Resources Cleaner
+
+on:
+  schedule:
+    - cron: "0 0 1 * *" # Run on the first day of every month
+  workflow_dispatch:
+    inputs:
+      daystokeep:
+        description: "days to keep a resource"
+        required: false
+        default: "30"
+
+env:
+  DAYS_TO_KEEP: 30
+
+jobs:
+  clean-resources:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 3
```

Comment:
Below 3 is fine since there would be a [limit to concurrent job](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits) across the whole workflow instead of a single workflow

Reply:
> its better to prioritize the integration test

Can you explain how limiting concurrency on job executions for this helps to prioritize integration tests? I'm trying to understand what we benefit from setting a max parallelism on this job.

Reply:
For example: if we have a **Free Github Plan** which only has 20 concurrent jobs, and assuming we will release CWAgent by using the GitHub workflow, there will be 2 scenarios:
*  If cleaner runs before CWAgent release, the `max-parallel`, which is 3, allows the cleaner to run 3 concurrent job. However, it would reduce the CWAgent release 's integration test to 17.
* If CWAgent release runs before cleaner, the cleaner would consume the remaining concurrent jobs which are allowed in the entire workflow, which could be 1, 2, or 3. 
Hope that make sense.

Reply:
If you remove the limit, what is the worst case?

Do we really need to be concerned about hitting the concurrent job limit?

Even if the limit is reached, won't the next job just wait for the next available "runner"?
(So worst case is just a delay)

Reply:
Worst case for this cleaning you are right though.